diff --git a/ld/linker.ld.S b/ld/linker.ld.S
index 40aefb92..f6281cdf 100644
--- a/ld/linker.ld.S
+++ b/ld/linker.ld.S
@@ -29,48 +29,8 @@ ENTRY(reset_handler)
 /* Define memory regions. */
 MEMORY
 {
-	/* RAM is always used */
-	ram (rwx) : ORIGIN = _RAM_OFF, LENGTH = _RAM
-
-#if defined(_ROM)
-	rom (rx) : ORIGIN = _ROM_OFF, LENGTH = _ROM
-#endif
-#if defined(_ROM1)
-	rom1 (rx) : ORIGIN = _ROM1_OFF, LENGTH = _ROM1
-#endif
-#if defined(_ROM2)
-	rom2 (rx) : ORIGIN = _ROM2_OFF, LENGTH = _ROM2
-#endif
-#if defined(_RAM1)
-	ram1 (rwx) : ORIGIN = _RAM1_OFF, LENGTH = _RAM1
-#endif
-#if defined(_RAM2)
-	ram2 (rwx) : ORIGIN = _RAM2_OFF, LENGTH = _RAM2
-#endif
-#if defined(_RAM3)
-	ram3 (rwx) : ORIGIN = _RAM3_OFF, LENGTH = _RAM3
-#endif
-#if defined(_RAM4)
-	ram4 (rwx) : ORIGIN = _RAM4_OFF, LENGTH = _RAM4
-#endif
-#if defined(_RAM5)
-	ram5 (rwx) : ORIGIN = _RAM5_OFF, LENGTH = _RAM5
-#endif
-#if defined(_CCM)
-	ccm (rwx) : ORIGIN = _CCM_OFF, LENGTH = _CCM
-#endif
-#if defined(_EEP)
-	eep (r) : ORIGIN = _EEP_OFF, LENGTH = _EEP
-#endif
-#if defined(_XSRAM)
-	xsram (rw) : ORIGIN = _XSRAM_OFF, LENGTH = _XSRAM
-#endif
-#if defined(_XDRAM)
-	xdram (rw) : ORIGIN = _XDRAM_OFF, LENGTH = _XDRAM
-#endif
-#if defined(_NFCRAM)
-	nfcram (rw) : ORIGIN _NFCRAM_OFF, LENGTH = _NFCRAM
-#endif
+	rom (rx) : ORIGIN = 0x08000000, LENGTH = 512K
+	ram (rwx) : ORIGIN = 0x20000000, LENGTH = 128K
 }
 
 /* Define sections. */
@@ -146,87 +106,6 @@ SECTIONS
 		_ebss = .;
 	} >ram
 
-#if defined(_CCM)
-	.ccm : {
-		_ccm = .;
-		*(.ccmram*)
-		. = ALIGN(4);
-		_eccm = .;
-	} >ccm
-#endif
-
-#if defined(_RAM1)
-	.ram1 : {
-		_ram1 = .;
-		*(.ram1*)
-		. = ALIGN(4);
-		_eram1 = .;
-	} >ram1
-#endif
-
-#if defined(_RAM2)
-	.ram2 : {
-		_ram2 = .;
-		*(.ram2*)
-		. = ALIGN(4);
-		_eram2 = .;
-	} >ram2
-#endif
-
-#if defined(_RAM3)
-	.ram3 : {
-		_ram3 = .;
-		*(.ram3*)
-		. = ALIGN(4);
-		_eram3 = .;
-	} >ram3
-#endif
-
-#if defined(_RAM4)
-	.ram4 : {
-		_ram4 = .;
-		*(.ram4*)
-		. = ALIGN(4);
-		_eram4 = .;
-	} >ram4
-#endif
-
-#if defined(_RAM5)
-	.ram5 : {
-		_ram5 = .;
-		*(.ram5*)
-		. = ALIGN(4);
-		_eram5 = .;
-	} >ram5
-#endif
-
-#if defined(_XSRAM)
-	.xsram : {
-		_xsram = .;
-		*(.xsram*)
-		. = ALIGN(4);
-		_exsram = .;
-	} >xsram
-#endif
-
-#if defined(_XDRAM)
-	.xdram : {
-		_xdram = .;
-		*(.xdram*)
-		. = ALIGN(4);
-		_exdram = .;
-	} >xdram
-#endif
-
-#if defined(_NFCRAM)
-	.nfcram : {
-		_nfcram = .;
-		*(.nfcram*)
-		. = ALIGN(4);
-		_enfcram = .;
-	} >nfcram
-#endif
-
 	/*
 	 * The .eh_frame section appears to be used for C++ exception handling.
 	 * You may need to fix this if you're using C++.
